File: ruby/client-config.rb
require 'bundler/setup'
require 'aws-sdk'

# build the client in place
creds  = Aws::Credentials.new(ENV['BUCKETEER_AWS_ACCESS_KEY_ID'], ENV['BUCKETEER_AWS_SECRET_ACCESS_KEY'])
s3     = Aws::S3::Client.new(credentials: creds, region: 'us-east-1')
key    = ARGV[0] || 'hello-client'
bucket = ENV['BUCKETEER_BUCKET_NAME']

s3.put_object(
  acl: 'private',
  bucket: bucket,
  key: key,
  body: 'world',
)

resp = s3.get_object(
  bucket: bucket,
  key: key
)

puts resp.body.read

File: node/client-config.js
var AWS = require('aws-sdk');

var s3  = new AWS.S3({
  accessKeyId: process.env.BUCKETEER_AWS_ACCESS_KEY_ID,
  secretAccessKey: process.env.BUCKETEER_AWS_SECRET_ACCESS_KEY,
  region: 'us-east-1',
});

var params = {
  Key:    process.argv[2] || 'hello-env',
  Bucket: process.env.BUCKETEER_BUCKET_NAME,
  Body:   new Buffer('Hello, node.js'),
};

s3.putObject(params, function put(err, data) {
  if (err) {
    console.log(err, err.stack);
    return;
  } else {
    console.log(data);
  }

  delete params.Body;
  s3.getObject(params, function put(err, data) {
    if (err) console.log(err, err.stack);
    else     console.log(data);

    console.log(data.Body.toString());
  });
});
